### PR TITLE
Prepare ana4 connector and add range from CI value

### PR DESCRIPTION
**Pull request by awirb**
* add option to vary within a range from a ci value read from shp file

### CODE REVIEW

File: avaframe/ana4Stats/probAna.py [hunk 1]
```diff
@@ -495,71 +506,287 @@ def createSampleWithPercentVariation(cfgProb, comMod, fileOverride=''):
...
+    # initialze lower and upper bounds required to get a sample for the parameter values
+    # numpy arrays required to do masking as lists don't work for a list indices
+    varValList = np.asarray([cfgStart['GENERAL'].getfloat(varPar) if varPar in staParameter else thValues[idx] for idx, varPar in enumerate(fullListOfParameters)])
+    fullValVar = np.asarray([float(valVariationValue[i]) if valVariationValue[i] != 'ci95' else np.nan for i in parentParameterId])
+    fullVarType = np.asarray([varType[i] for i in parentParameterId])
+    lowerBounds = np.asarray([None]*len(fullListOfParameters))
+    upperBounds = np.asarray([None]*len(fullListOfParameters))
+
+    # set lower and upper bounds depending on varType (percent, range, rangefromci)
+    lowerBounds[fullVarType == 'percent'] = (varValList[fullVarType == 'percent'] -
```
Comment (fso42) [inline avaframe/ana4Stats/probAna.py:661]:
```
for varType in ['percent','range ...]:
 lowerBounds[fullVarType == varType] ....
```

File: avaframe/ana4Stats/probAna.py [hunk 2]
```diff
@@ -495,71 +506,287 @@ def createSampleWithPercentVariation(cfgProb, comMod, fileOverride=''):
...
+        else:
+            parentParameterId.append(varParList.index(varPar))
+            fullListOfParameters.append(varPar)
+            staParameter.append(varPar)
+            thValues = np.append(thValues,np.asarray([None]))
+            ciValues = np.append(ciValues,np.asarray([None]))
+
+    # VERSION WITH USING MASKS--------------------------
+    # initialze lower and upper bounds required to get a sample for the parameter values
+    # numpy arrays required to do masking as lists don't work for a list indices
```
Comment (fso42) [inline avaframe/ana4Stats/probAna.py:653]:
```suggestion
   
    # initialize lower and upper bounds required to get a sample for the parameter values
    # numpy arrays are required to do masking as lists don't work for list indices
```

File: avaframe/ana4Stats/probAna.py [hunk 3]
```diff
@@ -495,71 +506,287 @@ def createSampleWithPercentVariation(cfgProb, comMod, fileOverride=''):
...
+    #         lB = varVal - float(valVariation)
+    #         uB = varVal + float(valVariation)
+    #     elif varType[parentIdx].lower() == 'rangefromci':
+    #         ciValue = ciValues[idx]
+    #         lB = varVal - float(ciValue)
+    #         uB = varVal + float(ciValue)
+    #     # update bounds
+    #     lowerBounds.append(lB)
+    #     upperBounds.append(uB)
+
```
Comment (fso42) [inline avaframe/ana4Stats/probAna.py:698]: remove

File: avaframe/ana4Stats/probAnaCfg.ini [hunk 4]
```diff
@@ -31,14 +31,15 @@ defaultComModuleCfg = True
 # list of parameters that shall be varied - separated by |
 varParList = relTh|mu
 
-variationType options: percent, range
+variationType options: percent, range, rangefromci
 # if percent: +- varParVariation percent of varParValue in varParSteps
 # if range: +- varParVariation of varParValue in varParSteps
+# if rangeFromCi: +- varParVariation but values taken from shape file ci value (only availabe for relTh, entTh, secondaryRelTh)
```
Comment (fso42) [inline avaframe/ana4Stats/probAnaCfg.ini:37]: There's a mismatch to line 39 rangeFromCi vs rangefromci

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/ana4Stats/probAnaCfg.ini (hunk 4 above)
Comment (awirb) [reply avaframe/ana4Stats/probAnaCfg.ini:37]: set to rangefromci in both cases

**Review by fso42 — APPROVED** [review]

**Review by fso42 — APPROVED** [review]

**Pull request merged**